src/index.tsx

```diff
@@ -46,6 +47,33 @@ type EventLog = {
   timestamp: Date;
 };
 
+type verifyServExtProps = {
+  api: SchedulerService;
+  trans: IRenderMime.TranslationBundle;
+};
+
+/**
+ * Call API to verify that the server extension is actually installed.
+ */
+async function verifyServExt(props: verifyServExtProps) {
+  try {
+    await props.api.getJobs({ max_items: 0 });
```

Comment:
How did you test this change ?  What is the behavior of `activatePlugin`  ? in my understanding, until this function completes, Jupyterlab is going to show the loading screen. 

It is not clear how sync vs async helps in solving this issue.

Reply:
Note that we are not awaiting [`verifyServerExtension` call inside `activatePlugin`](https://github.com/jupyter-server/jupyter-scheduler/pull/480/files#diff-0b5adbfe7b36e4ae2f479291e20152e33e940f7f265162d77f40f6bdb5da7405R181) vs having blocking `await api.getJobs({ max_items: 0 });` call there before. When `async` function is called without `await`, the code execution continues to the next line without waiting for the promise returned by the async function to resolve or reject, the main execution flow is not blocked, the asynchronous operation runs in the background and resolves whenever it finishes independent of the main execution flow.  

As mentioned in the PR description, to test:

1. Add a sleep statement like `await asyncio.sleep(20)` (20 seconds) to [JobHandler.get ](https://github.com/jupyter-server/jupyter-scheduler/blob/1b50f81f966c9e0146f91b1606fd78519857527e/jupyter_scheduler/handlers.py#L189) as so:

```python
class JobHandler(ExtensionHandlerMixin, JobHandlersMixin, APIHandler):
    @authenticated
    async def get(self, job_id=None):
        await asyncio.sleep(20)
        ...
```
2. Launch the extension with and without this PR, note that with this PR startup is not blocked

Reply:
@bhadrip please let me know if this answers your questions and resolves your concerns.

Reply:
The JupyterLab splash shows while any extensions are still activating. Removing the `await` statement from `activatePlugin()` will allow the function to complete regardless of how long `api.getJobs()` takes, meaning that Jupyter Scheduler will no longer be forcing the JupyterLab splash / loading screen to be shown after this PR.

While I agree testing would be nice here, testing should not block this PR, as the expected behavior is immediately apparent from code inspection.

Reply:
makes sense, thanks for the explanation.